Editors/SettingsEditor.cpp

```diff
@@ -0,0 +1,110 @@
+#include "SettingsEditor.h"
+#include "MainWindow.h"
+#include "ui_SettingsEditor.h"
+
+#include "codegen/Settings.pb.h"
+
+#include <QDebug>
+#include <QPushButton>
+
+using namespace buffers::resources;
+
+Q_DECLARE_METATYPE(buffers::SystemInfo);
+
+static std::string get_combo_system_id(const QComboBox* combo) {
+  auto data = combo->currentData();
+  auto subsystem = data.value<buffers::SystemInfo>();
+  return subsystem.id();
+}
+
+SettingsEditor::SettingsEditor(ProtoModel* model, QWidget* parent)
+    : BaseEditor(model, parent), ui(new Ui::SettingsEditor) {
+  ui->setupUi(this);
+
+  QPushButton* saveButton = ui->buttonBox->button(QDialogButtonBox::Save);
+  saveButton->setIcon(QIcon(":/actions/accept.png"));
+  connect(saveButton, &QPushButton::clicked, [=]() {
+    qDebug() << "hey";
+    Settings settings;
+    auto* api = settings.mutable_api();
+    api->set_target_audio(get_combo_system_id(ui->audioCombo));

```

Comment:
This is an extremely common pattern in this IDE that could be automated with those custom properties. Guess it's not a big deal, since at least it's clearly divided into modules.

Reply:
Sure, but this one is actually going away now that I have recursive mappers from merging the room editor pr. It will be replaced with the mapper stuff, which still isn't your ideal fix, but will be better.